### PR TITLE
Fixes "Bloodless monsters can still get bled by butchery, leave blood splatter on ranged hits"

### LINKED ISSUE #62113: Bloodless monsters can still get bled by butchery, leave blood splatter on ranged hits
### Describe the bug

Monsters with `bleed_rate 0` like late-stage zombies can still be bled on butchery, and leave blood splatters when hit by ranged attacks.

### Steps to reproduce

1. Spawn and kill a skeletal juggernaut
2. Butcher and bleed it to see it somehow still has blood
3. Spawn another and shoot at it with a big gun
4. Behold glorious blood splatters

### Expected behavior

Non-bleeding monsters not having blood, preferably *without* having to explicitly define separate `_no_blood` harvest lists.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2130 (21H2)
- Game Version: 0.F-12281-gac5fe8ca5b-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Standard combat test suite [standard_combat_test]
]

### Additional context

Mostly as a reminder for myself to take care of it sometime, but a good first C++ issue in any case.

### PR DESCRIPTION
**Pull request by alef**
#### Summary
Bugfixes "Bloodless monsters do no bleed #62113 "

#### Purpose of change
Fixes #62113

#### Describe the solution
Fix the C++ code on the way during debug sessions in VS/SDL.

#### Describe alternatives you've considered
None

#### Testing
Following the steps in #62113.

#### Additional context
From `mtype.h`: _Bleed rate in percent, 0 makes the monster immune to bleeding_
From `MONSTERS.md`: _0 translates to bleed immunity_

### CODE REVIEW

**Comment by alef** [comment]
[map::add_splatter](https://github.com/alef/Cataclysm-DDA/blob/2af53d83c3714e50f59fdbbdc998eb9b43e0a9eb/src/map.cpp#L6236) should now receive an `fd_null`, which should not have `is_splattering`.

File: src/ballistics.cpp [hunk 1]
```diff
@@ -442,10 +442,13 @@ dealt_projectile_attack projectile_attack( const projectile &proj_arg, const tri
             // Critter can still dodge the projectile
             // In this case hit_critter won't be set
             if( attack.hit_critter != nullptr ) {
-                const size_t bt_len = blood_trail_len( attack.dealt_dam.total_damage() );
-                if( bt_len > 0 ) {
-                    const tripoint &dest = move_along_line( tp, trajectory, bt_len );
-                    here.add_splatter_trail( critter->bloodType(), tp, dest );
+                const auto blood_type = critter->bloodType();
+                if( blood_type ) {
```
Comment (alef) [inline src/ballistics.cpp:446]: This is not really needed, it just avoids to do the computation.

File: src/ballistics.cpp [hunk 2]
```diff
@@ -442,10 +442,13 @@ dealt_projectile_attack projectile_attack( const projectile &proj_arg, const tri
             // Critter can still dodge the projectile
             // In this case hit_critter won't be set
             if( attack.hit_critter != nullptr ) {
-                const size_t bt_len = blood_trail_len( attack.dealt_dam.total_damage() );
-                if( bt_len > 0 ) {
-                    const tripoint &dest = move_along_line( tp, trajectory, bt_len );
-                    here.add_splatter_trail( critter->bloodType(), tp, dest );
+                const auto blood_type = critter->bloodType();
```
Comment (dseguin) [inline src/ballistics.cpp:445]:
```suggestion
                const field_type_id blood_type = critter->bloodType();
```

File: src/game.cpp [hunk 3]
```diff
@@ -8747,9 +8747,10 @@ static void butcher_submenu( const std::vector<map_stack::iterator> &corpses, in
                         corpses[index]->has_flag( flag_FIELD_DRESS_FAILED ) ) ) {
                     has_organs = true;
                 }
-                if( entry.type == harvest_drop_blood && !( corpses[index]->has_flag( flag_QUARTERED ) ||
-                        corpses[index]->has_flag( flag_FIELD_DRESS ) ||
-                        corpses[index]->has_flag( flag_FIELD_DRESS_FAILED ) || corpses[index]->has_flag( flag_BLED ) ) ) {
+                if( entry.type == harvest_drop_blood && dead_mon->bleed_rate &&
```
Comment (dseguin) [inline src/game.cpp:8750]:
```suggestion
                if( entry.type == harvest_drop_blood && dead_mon->bleed_rate > 0 &&
```

File: src/mtype.cpp [hunk 4]
```diff
@@ -153,9 +153,11 @@ std::vector<std::string> mtype::species_descriptions() const
 
 field_type_id mtype::get_bleed_type() const
 {
-    for( const species_id &s : species ) {
-        if( !s->bleeds.is_empty() ) {
-            return s->bleeds;
+    if( bleed_rate ) {
```
Comment (dseguin) [inline src/mtype.cpp:156]:
```suggestion
    if( bleed_rate > 0 ) {
```

**Review by dseguin — APPROVED** [review]
LGTM, just a few minor changes to make things more explicit.

**Pull request merged**